src/coreEnforcer.ts

```diff
@@ -278,7 +280,13 @@ export class CoreEnforcer {
       throw new Error('model is undefined');
     }
 
-    const expression = compileAsync(expString);
+    let expression;
+    if (this.matcherMap.has(expString)) {
+      expression = await this.matcherMap.get(expString);
```

Comment:
```suggestion
      expression = this.matcherMap.get(expString);
```

There's no need to await